pandas/_libs/tslibs/strptime.pyx

```diff
@@ -504,7 +514,14 @@ def array_strptime(
             result_timezone[i] = tz
 
         except (ValueError, OutOfBoundsDatetime) as ex:
-            ex.args = (f"{str(ex)}, at position {i}",)
+            ex.args = (
+                f"{str(ex)}, at position {i}. You might want to try:\n"
+                "    - passing ``format='ISO8601'`` if your strings are "
+                "all ISO8601 but not necessarily in exactly the same format;\n"
+                "    - passing ``format='mixed'``, and the format will be "
+                "inferred for each element individually. "
+                "You might want to use ``dayfirst`` alongside this.",
```

Comment:
We might still want to mention the suggestion to pass a specific `format` first?

For example, if an error occurs because the format is ambiguous based on the first value (dayfirst/daylast), and was inferred incorrectly:

```
pd.to_datetime(["12/01/2012", "13/01/2012"])
...
ValueError: time data "13/01/2012" doesn't match format "%m/%d/%Y", at position 1. You might want to try:
    - passing ``format='ISO8601'`` if your strings are all ISO8601 but not necessarily in exactly the same format;
    - passing ``format='mixed'``, and the format will be inferred for each element individually. You might want to use ``dayfirst`` alongside this.
```

Here, I think the best is still to provide a manual `format="%d/%m/%Y"` (or to specify `dayfirst=True`, then the inference would work for this data)

Reply:
(also, a fomatting nit: since this is plain text (and not rst that will be rendered), no need for double backticks, I would use single backticks which is a bit less visually distracting)

Reply:
good point, thanks!